libfreerdp/core/peer.c

```diff
@@ -960,13 +961,26 @@ static state_run_t peer_recv_callback_internal(rdpTransport* transport, wStream*
 		case CONNECTION_STATE_MULTITRANSPORT_BOOTSTRAPPING_REQUEST:
 			if (settings->SupportMultitransport)
 			{
-				if (!multitransport_server_send_request(rdp->multitransport))
-					ret = STATE_RUN_FAILED;
-				else
+				/* TODO: this counter shall end up in a listener or server instance */
+				static UINT32 reqId = 0;
```

Comment:
Well this counter is something kinda global, we could say bound to the listener component. 

The global scenario is an accepted TCP peer will grab one of these requestId, will send the packet with id and cookie over TCP. 
After a while we're gonna have an incoming UDP connection (on a UDP listener or the UDP part of the listener), establishing the UDP transport, and in the final packet it will send to the corresponding requestId and cookie. When we receive that, we must inquiry current TCP connections (that mean that the listener or another component must keep track of these), and if we have a reqId/cookie match, then the incoming UDP transport will be set as UDP part of the corresponding RDP session.

Reply:
So to answer the comment, the cookie could/should be generated in `multitransport_server_request`, but for the request counter we must have a larger view.

Reply:
Ok moved the logic in multitransport_server_request